### ISSUE TITLE
Print warning instead of error when using async

### ISSUE THREAD

**Issue by ccjruemenapf** [erikras/redux-form#2827]
Hi Folks,
i would be nice, if there will be a possibility to create warnings instead of errors when using the async validation. 
And if i am not wrong only asyncErrors exists and no asyncWarnings.

We want to build async uniqueness checks and show a warning not an error, if the endpoint sad - Nope.

Is there a chance or a plan to get a feature like this?

Greetings and thank you
Jan

**Comment by griffinmichl** [comment]
Agreed on this enhancement! I want it in an app I'm working on where there's a URL input field. I'd like to be able to warn users that we couldn't resolve the URL they input.  
  
I'm gonna work on this if no one else is.

**Comment by kentandersen** [comment]
It would be great if this could have the same api as [asyncValidate](https://redux-form.com/6.7.0/examples/asyncValidation/).
For example:
```
reduxForm({
  form: 'asyncWarn',
  validate,
  asyncValidate,
  asyncValidateBlurFields: ['username'], // renamed form asyncBlurFields to prevent confusion
  asyncWarn,
  asyncWarnBlurFields: ['username']
})(AsyncWarnForm)
```

`asyncBlurFields` should be an alias to `asyncValidateBlurFields` to prevent breaking backwards compatibility

**Comment by SimonPNorra** [comment]
Hey @erikras and @griffinmichl,

do you have any idea for a workaround to support warnings in asyncValidations? We really need it in our current project and we are struggling finding a solution.

Thanks a lot.

Regards,
Simon

**Comment by terolepisto** [comment]
Has this have any progress? Is there a way to "push" any warnings otherway, like we have ability to change form values with this.props.change("foo","bar") ?

Like this.props.warning("foo","Double check Your Foo!") ?

**Comment by AndyCJ** [comment]
Has anyone found a workaround for this?

As @terolepisto says, if there's a way to manually add warnings to a form then external code could handle the async validation warning check and add to warning to the form when it returns.

**Comment by terolepisto** [comment]
I've now used reducers to pass errors to the components which means I have
to pass them as props deep deep deep to the actual input fields and forms.

---
Tero Lepistö

pe 26. tammikuuta 2018 klo 13.35 Andy Jump <notifications@github.com>
kirjoitti:

**Comment by AndyCJ** [comment]
Thank you!

I hadn't thought of side-stepping redux-forms entirely, but that makes a lot of sense.

That gives me a workaround to keep me going.

I've posted this as a question to Stack Overflow, and it'd be of value to have your suggestion on there as an answer: https://stackoverflow.com/questions/48461100/how-can-i-show-a-redux-form-warning-as-a-result-of-async-validation

If SO isn't your thing then no worries, I'll add it as an answer in a few days with a link attributing it to here.

**Comment by ughitsaaron** [comment]
The async validation callback passes in `dispatch` as a second parameter which you can use to dispatch actions that create your own custom warnings.

**Comment by szaboat** [comment]
> Agreed on this enhancement! I want it in an app I'm working on where there's a URL input field. I'd like to be able to warn users that we couldn't resolve the URL they input.
> 
> I'm gonna work on this if no one else is.

What exactly agreed? I've bumped into this issue and could not find a proper solution and thinking about implementing it. There were two options discussed
* have a new api and expose the `asyncWarn` and `asyncWarnBlurFields` 
* return a promise in the asyncValidation with `warnings` in it
What would be the preferred way? Thanks!

**Comment by jwanglof** [comment]
@szaboat I think that new APIs would be preferable with the reason you express, the validations are separated for the synchronous validation 👍
